app/src/features/apiClient/helpers/apiClientExecutor/apiClientExecutor.ts

```
@@ -104,44 +105,31 @@ export class RequestExecutor {
     this.internalFunctions = internalFunctions;
   }
 
-  async executePreRequestScript() {
+  async executePreRequestScript(callback: (state: any) => Promise<void> = () => Promise.resolve()) {
```

In this context, having a default callback hides complexity that shouldn't be hidden. If someone wants to pass an empty callback, let them be upfront about it.